ui/library/src/components/DocumentWrapper.tsx

```diff
@@ -51,11 +55,31 @@ export const DocumentWrapper: React.FunctionComponent<Props> = ({
     setIsLoading(false);
   }, []);
 
+  const onItemClicked = (param: any): void => {
+    if (!pdfDocProxy) {
+      return;
+    }
+
+    // Scroll to the destination of the item
+    pdfDocProxy.getDestination(param.dest).then(destArray => {
+      if (!destArray) {
+        return;
+      }
+
+      const [ref, , , bottomPoints] = destArray;
+      pdfDocProxy.getPageIndex(new Ref(ref)).then(refInfo => {
+        scrollToPosition(parseInt(refInfo.toString()), 0, bottomPoints, rotation);
+      });
+    });
+  };
+
   return (
     <Document
       options={{ cMapUrl: 'cmaps/', cMapPacked: true }}
       onLoadError={onPdfLoadError}
       onLoadSuccess={onPdfLoadSuccess}
+      // @ts-ignore: an object with { dest, pageIndex, pageNumber } is passed to the handler function but tslint thinks it should be { pageNumber }
+      onItemClick={onItemClicked}
```

Comment:
Ah this is where I was stuck. This callback wasn't getting called. Was that the issue with the react version?

Reply:
I think the issue was caused by incorrect argument passing to the linkService function, which is the same as this one https://github.com/wojtekmaj/react-pdf/issues/850. The author fixed it after `v5.4.1` that makes me believe I should update the `react-pdf` package 😆

Reply:
Question: Would this also be called with the react-pdf Outline component was used? Doesn't look like our outline component is a wrapper for the Outline component in react-pdf. Something we should look at later?

Reply:
Yes, the react-pdf Outline component has its own `onItemClick` property. By the time I was implementing TOC, react-pdf didn't pass destination information to `onItemClick` in its Outline component, which made me have to create a new one for ourselves. But even the new version of react-pdf passes the information, I think it's still better for us to create our own Outline component for adding expand buttons and showing the number of annotations in M3.